pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1601,3 +1614,120 @@ def _check_return_at_the_end(self, node):
             # return None"
             elif isinstance(last.value, astroid.Const) and (last.value.value is None):
                 self.add_message("useless-return", node=node)
+
+    def _check_unnecessary_dict_indexing(
+        self, node: Union[astroid.For, astroid.Comprehension]
+    ) -> None:
+        """Add message when accessing dict values by index lookup."""
+        # Verify that we have a .items() call and
+        # that the object which is iterated is used as a subscript in the
+        # body of the for.
+        # Is it a proper items call?
+        if (
+            isinstance(node.iter, astroid.Call)
+            and isinstance(node.iter.func, astroid.Attribute)
+            and node.iter.func.attrname == "items"
+        ):
+            inferred = utils.safe_infer(node.iter.func)
+            if not isinstance(inferred, astroid.BoundMethod):
+                return
+            iterating_object_name = node.iter.func.expr.as_string()
+
+            # Verify that the body of the for loop uses a subscript
+            # with the object that was iterated. This uses some heuristics
+            # in order to make sure that the same object is used in the
+            # for body.
+
+            children = (
```

Comment:
```suggestion
            children: Tuple[astroid.node_classes.NodeNG] = (
```
That's only necessary since `astroid` isn't yet fully typed. You will also need to add imports for `Tuple` and `astroid.node_classes`.

Reply:
I'm not too sure I follow this part.. could you elaborate a bit further on why ``Tuple`` and what this is for? sorry for the trouble

Reply:
fixed 0b414d3a, but I kinda wanna know why so leaving this comment unresolved for now

Reply:
Not completely sure why I suggested it 🤦🏻 After taking another look, I now see that `Tuple[astroid.node_classes.NodeNG]` is probably wrong. Might be best to revert this one and just leave the type empty.

The real one would probably be something like this
```py
Union[List[NodeNG], None, Generator[NodeNG, None, None]]
```